epochStart/bootstrap/process_test.go

```diff
@@ -681,30 +1583,429 @@ func getNodesConfigMock(numOfShards uint32) sharding.GenesisNodesSetupHandler {
 	return nodesConfig
 }
 
-func TestRequestAndProcessing(t *testing.T) {
+func TestRequestAndProcessing_ShouldFail(t *testing.T) {
```

Comment:
I've just put `requestAndProcessFromStorage` and `requestAndProcessing` through a compare tool and they are absolutely identical except for 2 things:

1. One gets its meta header from storage:
 -   `sesb.syncedHeaders, err = sesb.syncHeadersFromStorage(sesb.epochStartMeta, sesb.destinationShardAsObserver)` , while the other syncs it directly:
 -       `e.syncedHeaders, err = e.syncHeadersFrom(e.epochStartMeta)`

2. `requestAndProcessing` does an extra cache clearing: ` e.dataPool.TrieNodes().Clear()`

Isn't there any way to extract all this common code to a separate func and only test the common code?
Otherwise, not only that the productive code is duplicated, but these tests are as well (which require even more lines of code)
This way, this test + many other tests would be removed

Reply:
:+1: will consider this, and see if it's worth to do in this PR, or we can think of this for another one